src/main/java/uk/ac/ebi/intact/app/internal/ui/components/query/advanced/tests/MIQLParserTest.java

```diff
@@ -0,0 +1,144 @@
+package uk.ac.ebi.intact.app.internal.ui.components.query.advanced.tests;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import uk.ac.ebi.intact.app.internal.ui.components.query.advanced.Field;
+import uk.ac.ebi.intact.app.internal.ui.components.query.advanced.MIQLParser;
+import uk.ac.ebi.intact.app.internal.ui.components.query.advanced.parser.components.Rule;
+import uk.ac.ebi.intact.app.internal.ui.components.query.advanced.parser.components.RuleComponent;
+import uk.ac.ebi.intact.app.internal.ui.components.query.advanced.parser.components.RuleSet;
+
+import java.util.Objects;
+
+public class MIQLParserTest {
```

Comment:
Tests should be separate from the source code.
We have:
- src
  - main
    - java
      - uk.ac.ebi.intact.app.internal
    - resources
  - test
    - java
      - uk.ac.ebi.intact.app.internal
    - resources

Reply:
Check other repos to see where we put the test files